src/models/user.model.ts

```diff
@@ -0,0 +1,32 @@
+import { Schema, Document, model, PassportLocalSchema } from 'mongoose';
+import passportLocalMongoose from 'passport-local-mongoose';
```

Comment:
Can we keep the passport dependency separate from the models? Because if we decide to separate auth service, or switch out to something else, we shouldn't have to change our models.

Reply:
With respect to this particular dependency, we can't cauz that would just mean that we have to manually add this plugin wherever we use the user model.

Reply:
In case we switch to something else, we won't really have to make any changes in the model itself. The only thing this dependency does is that it adds in the necessary fields for us.

Reply:
Alright, sounds good!